zmon_worker_monitor/builtins/plugins/dns_.py

```diff
@@ -43,3 +44,26 @@ def resolve(self, host=None):
         except Exception, e:
             result = 'ERROR: ' + str(e)
         return result
+
+    def query(self, host=None, recordtype=None, nameserver=None):
+        if not self.host and not host:
+            raise ConfigurationError('DNS wrapper improperly configured. Host is required!')
+
+        if not recordtype:
+            raise ConfigurationError('DNS wrapper improperly configure. Recordtype is required!')
+
+        try:
+            resolver = dns.resolver.Resolver()
+            # Allow to use custom nameserver
+            if nameserver:
+                resolver.nameservers = nameserver
+
+            query = resolver.query(host, recordtype)
+            result = []
+
+            for response in query:
+                result.append(str(response).replace('"', ''))
+
+        except Exception, e:
```

Comment:
I think it would be better to let the exception bubble, since it is handled by ZMON (visually). Also, the checks could have their own handling of ``try ... except``.